src/main/java/uk/gov/hmcts/juror/api/moj/repository/letter/court/ExcusalRefusalLetterListRepositoryImpl.java

```diff
@@ -1,79 +1,193 @@
 package uk.gov.hmcts.juror.api.moj.repository.letter.court;
 
-import com.querydsl.core.types.dsl.CaseBuilder;
+import com.querydsl.core.Tuple;
 import com.querydsl.jpa.impl.JPAQuery;
 import com.querydsl.jpa.impl.JPAQueryFactory;
 import io.micrometer.common.util.StringUtils;
 import jakarta.persistence.EntityManager;
 import jakarta.persistence.PersistenceContext;
+import uk.gov.hmcts.juror.api.moj.domain.QJuror;
+import uk.gov.hmcts.juror.api.moj.domain.QJurorHistory;
+import uk.gov.hmcts.juror.api.moj.domain.QJurorPool;
+import uk.gov.hmcts.juror.api.moj.domain.QJurorStatus;
+import uk.gov.hmcts.juror.api.moj.domain.QPoolRequest;
 import uk.gov.hmcts.juror.api.moj.domain.letter.CourtLetterSearchCriteria;
 import uk.gov.hmcts.juror.api.moj.domain.letter.court.ExcusalRefusedLetterList;
-import uk.gov.hmcts.juror.api.moj.domain.letter.court.QExcusalRefusedLetterList;
+import uk.gov.hmcts.juror.api.moj.enumeration.HistoryCodeMod;
 
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Comparator;
 import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
 
 public class ExcusalRefusalLetterListRepositoryImpl implements IExcusalRefusalLetterListRepository {
 
     @PersistenceContext
     EntityManager entityManager;
 
-    private static final QExcusalRefusedLetterList EXCUSAL_REFUSED_LETTER_LIST =
-        QExcusalRefusedLetterList.excusalRefusedLetterList;
-
     @Override
+    @SuppressWarnings("PMD.CognitiveComplexity")
```

Comment:
In stead of suppressing this can we split this method out into several smaller methods?

Reply:
have broken method into sub methods